### PR TITLE
chore: Elaborate on all TODOs

### LINKED ISSUE #667: Elaborate on or implement all TODOs in Test Suite
Ensure they are dealt with or adequately documented

### PR DESCRIPTION
**Pull request by lukehesluke**
Closes #667 

The approach I've taken:

* Particularly important issues or recurring issues have been transformed into issues (which are linked to from the TODO)
* Some very simple TODOs have just been done 🙌 
* Other TODOs have simply been elaborated with any context that I have or can find to help others be able to implement them in future

### CODE REVIEW

File: packages/openactive-integration-tests/test/features/approval/minimal-proposal/implemented/accept-proposal-book-test.js [hunk 1]
```diff
@@ -70,13 +70,12 @@ FeatureHelper.describeFeature(module, {
...
   });
   FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(bookRecipe, () => {
-    // TODO does validator already check that orderProposalVersion is of form {orderId}/versions/{versionUuid}?
+    // TODO Validator should check this: https://github.com/openactive/data-model-validator/issues/449
     it('should include an orderProposalVersion, of the form {orderId}/versions/{versionUuid}', () => {
       const { uuid } = defaultFlowStageParams;
       expect(bookRecipe.p.getOutput().httpResponse.body).to.have.property('orderProposalVersion')
         .which.matches(RegExp(`${uuid}/versions/.+`));
     });
-    // TODO does validator check that full Seller details are included in the seller response?
```
Comment (lukehesluke) [inline packages/openactive-integration-tests/test/features/approval/minimal-proposal/implemented/accept-proposal-book-test.js:79]: it does

File: packages/openactive-integration-tests/test/features/details-capture/customer-details-capture-non-essential/implemented/non-essential-customer-details-reflected-test.js [hunk 2]
```diff
@@ -26,6 +26,9 @@ function (configuration, orderItemCriteriaList, featureIsImplemented, logger, de
   FlowStageUtils.describeRunAndCheckIsSuccessfulAndValid(c2, () => {
     it('should include expected customer details', () => {
       const apiResponseJson = c2.getStage('c2').getOutput().httpResponse.body;
+      if (defaultFlowStageParams.customer['@type'] !== 'Person') {
```
Comment (lukehesluke) [inline packages/openactive-integration-tests/test/features/details-capture/customer-details-capture-non-essential/implemented/non-essential-customer-details-reflected-test.js:29]: Required due to improving the `customer` type to also support `Organization`

File: packages/openactive-integration-tests/test/helpers/flow-stages/book-recipe.js [hunk 3]
```diff
@@ -6,7 +6,6 @@
  * @typedef {import('./assert-opportunity-capacity').AssertOpportunityCapacityFlowStageType} AssertOpportunityCapacityFlowStageType
  */
 
-// TODO construct this class using the generic FlowStageRun (e.g. with inheritance), rather than have it be a bespoke thing
```
Comment (lukehesluke) [inline packages/openactive-integration-tests/test/helpers/flow-stages/book-recipe.js:9]: It doesn't really seem that important now that it's been a while

File: packages/openactive-integration-tests/test/templates/b-req.js [hunk 4]
```diff
@@ -399,19 +406,13 @@ function createMissingPaymentReconciliationDetailsBReq(data) {
  */
 function createIncorrectReconciliationDetails(data) {
   const req = createStandardPaidBReq(data);
-  // @ts-ignore // TODO: Figure out why req.payment isn't recognised here?!
```
Comment (lukehesluke) [inline packages/openactive-integration-tests/test/templates/b-req.js:402]: Type was updated above to fix this issue

**Review by civsiv — APPROVED** [review]

**Pull request merged**